File: examples/documents/share-document.swift
import JsBaoClient

// Share a document by user id, by email, or with an entire group.
func shareDocument(client: JsBaoClient, documentId: String) async throws {
  // #region example
  // By user ID
  _ = try await client.documents.updatePermissions(
    documentId: documentId,
    params: ["userId": "user-abc", "permission": "read-write"]
  )

  // By email — works whether or not the recipient is a member yet
  _ = try await client.documents.updatePermissions(
    documentId: documentId,
    params: ["email": "colleague@example.com", "permission": "read-write"]
  )

  // With a group
  _ = try await client.documents.grantGroupPermission(
    documentId: documentId,
    params: ["groupType": "team", "groupId": "engineering", "permission": "read-write"]
  )
  // #endregion example
}

File: examples/documents/share-document.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Share a document by user id, by email, or with an entire group.
export async function shareDocument(client: JsBaoClient, documentId: string) {
  // #region example
  // By user ID
  await client.documents.updatePermissions(documentId, {
    userId: "user-abc",
    permission: "read-write",
  });

  // By email — works whether or not the recipient is a member yet
  await client.documents.updatePermissions(documentId, {
    email: "colleague@example.com",
    permission: "read-write",
  });

  // With a group
  await client.documents.grantGroupPermission(documentId, {
    groupType: "team",
    groupId: "engineering",
    permission: "read-write",
  });
  // #endregion example
}
